### ISSUE TITLE
What are batch1 and batch2 (input1 and input2)?

### ISSUE THREAD

**Issue by mingqiu** [williamleif/GraphSAGE#48]
Hi there,

I'm trying to understand the model but am a little bit confused here.
`        
self.inputs1 = placeholders["batch1"]
self.inputs2 = placeholders["batch2"]
`
I'm wondering why there are two inputs and two outputs? I saw the final node embeddings are save in model.output1. So then what are output2?

Thanks,
Serena

**Comment by ghostHBL** [comment]
The inputs2 is the neighbor node of inputs1, and the output2 is the representations for this neighbor node, the inputs2 and output2 are only used in the training process. As the paper said, "The
graph-based loss function encourages nearby nodes to have similar representations", we need the outputs to compute the loss.

**Comment by RexYing** [comment]
The interpretation is correct. Thanks for explaining!
